client/src/components/EditProfile/PaymentTab/PaymentTab.tsx

```diff
@@ -0,0 +1,125 @@
+        cardExpDate: values.cardExpDate,
+        cardCVC: values.cardCVC,
+        cardType: ['visa', 'mastercard'][Math.floor(Math.random() * 2)],
+      });
+      formik.resetForm();
+      setShowCardInput(!showCardInput);
+    },
+  });
+
+  return (
+    <Box minHeight="50vh" padding="45px 40px" display="flex" flexDirection="column" alignItems="center">
+      <Typography className={classes.paymentHeader}>Payment Methods</Typography>
+      {savedCards.length > 0 && !showCardInput && (
+        <Box width="100%" marginBottom="30px">
+          <Typography className={classes.savedText}>Saved Payment Profiles: </Typography>
+          <Box display="flex" width="100%" height="300px" overflow="scroll">
+            {savedCards.map((card) => {
+              return <CreditCard card={card} key={card.cardNumber} />;
+            })}
+          </Box>
+        </Box>
+      )}
+      {!showCardInput && (
+        <Button onClick={(e) => setShowCardInput(!showCardInput)} className={classes.showCardInputButton}>
+          Add new payment profile
+        </Button>
+      )}
+      {showCardInput && (
+        <Box width="100%">
+          <form className={classes.cardForm} onSubmit={formik.handleSubmit}>
+            <Typography className={classes.cardFormLabel}>Name</Typography>
+            <OutlinedInput
+              name="name"
+              value={formik.values.name}
+              onChange={formik.handleChange}
+              type="text"
+              autoComplete="cc-name"
+              placeholder="Name On Card"
+              fullWidth
+            />
+            <Typography className={classes.cardFormLabel}>Credit Card Number</Typography>
+            <OutlinedInput
+              name="cardNumber"
+              value={formik.values.cardNumber}
+              onChange={formik.handleChange}
+              type="text"
+              autoComplete="cc-number"
+              placeholder="9999 9999 9999 9999"
+              fullWidth
+            />
+            <Grid className={classes.cardFormGrid}>
+              <Box>
+                <Typography className={classes.cardFormLabel}>Expiration Date</Typography>
+                <OutlinedInput
+                  name="cardExpDate"
+                  value={formik.values.cardExpDate}
+                  onChange={formik.handleChange}
+                  type="text"
+                  autoComplete="cc-exp"
+                  placeholder="MM/YY"
+                  fullWidth
+                />
+              </Box>
+              <Box>
+                <Typography className={classes.cardFormLabel}>Security Code</Typography>
+                <OutlinedInput
+                  name="cardCVC"
+                  value={formik.values.cardCVC}
+                  onChange={formik.handleChange}
+                  type="password"
+                  autoComplete="cc-csc"
+                  placeholder="CVC"
+                  fullWidth
+                />
+              </Box>
+            </Grid>
```

Comment:
We could implement these with Stripe js instead, for example a [`CardElement`](https://stripe.com/docs/js/elements_object/create_element?type=card) or a [`PaymentElement`](https://stripe.com/docs/js/element/payment_element) might be useful ([they are also present in React Stripe](https://stripe.com/docs/stripe-js/react#available-element-components)). [This](https://stripe.com/docs/payments/payment-element) is a decent starting point in the docs as well 👍🏼